ext/gd/gd.c

```diff
@@ -1413,15 +1395,8 @@ static int _php_image_type(zend_string *data)
 		}
 	}
 
-	gdIOCtx *io_ctx;
-	io_ctx = gdNewDynamicCtxEx(8, ZSTR_VAL(data), 0);
-	if (io_ctx) {
-		if (_php_ctx_getmbi(io_ctx) == 0 && _php_ctx_getmbi(io_ctx) >= 0) {
```

Comment:
Now looking at it, this line weirds me out a bit or maybe I just do not get it.

Reply:
An MBI is an integer with a representation similar to UTF-8 characters (if high bit is set, the next byte also counts). The line reads two MBIs; the first one needs to be zero (that is important), and the second one greater than or equal to zero (what it is supposed to be anyway; there is no provision for negative MBIs).

Reply:
oh gotcha thx !